PR against issue 64

chughpiyush opened the pull request:
## Summary of the changes / Why this is an improvement
- Vulnerability detected in a particular version in build.gradle - Issue 64 - https://github.com/crate/crate-java-testing/issues/64

Primarily had to update apache.commons-compress version to 1.4.1
But also updated Junit version to 4.12 and gson version to 2.8.6
All tests were passing except for the one, I have created another PR.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

chughpiyush commented:
> @chughpiyush Thank you for that contribution (and catching this!)
> Would you mind creating a dedicated commit for the new dependencycheck plugin/task addition?
> Also it may worth mentioning at the `DEVELOP.rst` I guess.

This one - https://github.com/crate/crate-docs-theme/blob/master/DEVELOP.rst ?

seut commented:
> This one - https://github.com/crate/crate-docs-theme/blob/master/DEVELOP.rst

This one https://github.com/crate/crate-java-testing/blob/master/DEVELOP.rst ;-)

chughpiyush commented:
> > This one - https://github.com/crate/crate-docs-theme/blob/master/DEVELOP.rst
> 
> This one https://github.com/crate/crate-java-testing/blob/master/DEVELOP.rst ;-)

Missed that one 😅 thanks!
Am I allowed to do changes as described in `Preparing a Release` (I think it could be 0.7.1)?

chughpiyush commented:
> @chughpiyush Thank you for that contribution (and catching this!)
> Would you mind creating a dedicated commit for the new dependencycheck plugin/task addition?
> Also it may worth mentioning at the `DEVELOP.rst` I guess.

It's done, please review.

seut commented on DEVELOP.rst:
```diff
@@ -23,6 +23,15 @@ Run the unit tests like so::
+Vulnerability Check
+===================
+
+You should run this command to check if any dependency has a vulnerability::
```
```suggestion
Dependencies can be checked for known vulnerabilities by running::
```

chughpiyush replied:
This has been fixed.

seut commented on DEVELOP.rst:
```diff
@@ -23,6 +23,15 @@ Run the unit tests like so::
+
+    $ ./gradlew dependencyCheckAnalyze
+
+So you should run it whenever you're doing any dependency changes in build.gradle
```
I think another good usage of this is running it periodically, vulnerabilities are mostly found after some version was released and used afaik.
Maybe we should run this task always as a pre-compile task instead similar on how we do style checks?

seut replied:
Ah hm I think this task may run quite long (it downloads the NVD database) so running it as a hook may not be acceptable. Maybe we should just leave this advice/sentence  out?

chughpiyush replied:
Removed this advice (and yes you're right, it should be run on-demand)

seut commented:
@chughpiyush I've added another commit to update all dependencies to the current latest stable versions as of today. Or is there a concrete reason why you didn't do so (you pinned older versions)?

chughpiyush commented:
> @chughpiyush I've added a commit to update all dependencies to the current latest stable versions as of today. Or is there a concrete reason why you didn't do so (you pinned older versions)?

I remember I got a compile time error when I updated `randomizedTestingVersion` to latest and I didn't want to unnecessarily update something that I thought I shouldn't (my first PR for crate 😅). Your commit looks good to me and I think we can proceed to go ahead.
